docs/components/MarkdownPage.js

```diff
@@ -0,0 +1,97 @@
+// @flow strict
+
+// ignoring: since we do in fact want to render each component md block again
+// eslint-disable-next-line eslint-comments/disable-enable-pair
+/* eslint-disable react/no-unstable-nested-components */
+import { Text, Box } from 'gestalt';
+import { MDXProvider } from '@mdx-js/react';
+import { type Node } from 'react';
+
+import Page from './Page.js';
+import PageHeader from './PageHeader.js';
+import MainSection from './MainSection.js';
+
+import { MAX_WIDTH } from './MainSectionSubsection.js';
+
+type Props = {|
+  children: Node,
+  meta: {|
+    title: string,
+    badge: 'pilot' | 'deprecated',
+    component?: boolean,
+    description: string,
+  |},
+  pageSourceUrl?: string,
+|};
+
+export default function MarkdownPage({ children, meta, pageSourceUrl }: Props): Node {
+  const components = {
+    pre: (props) => <MainSection.Card defaultCode={props.children.props.children} />,
+
+    h2: (props) => (
+      <Box marginTop={12} marginBottom={4}>
+        <MainSection name={props.children} />
+      </Box>
+    ),
+    h3: (props) => (
+      <Box
+        ref={(node) => {
+          // eek, hacky. Getting around the .Markdown > h3 coloring css
+          if (node) {
+            node.classList.add('mdx-header');
+          }
+        }}
+      >
+        <MainSection.Subsection title={props.children} />{' '}
+      </Box>
+    ),
+    Card: (props) => {
+      const newProps = { ...props };
+      newProps.description = undefined;
+      return <MainSection.Card {...newProps} />;
+    },
+    Code: (props: {| removeMarginBottom: boolean, children: string| null |}) => {
+      const newProps = { ...props };
+      newProps.children = null;
+      newProps.removeMarginBottom = false;
+      return (
+        <MainSection.Card
+          {...newProps}
+          defaultCode={props.children || ""}
+          removeMarginBottom={!('removeMarginBottom' in props) ? true : props.removeMarginBottom}
+        />
+      );
+    },
+    Group: (props) => <Box marginBottom={12}>{props.children}</Box>,
```

Comment:
If we aren't doing components, we might not need these ones no?

Just basic markdown

Reply:
Could you also add a  "Create a markdown page" in Development section to get an idea of the easiness to the general user?
https://gestalt.pinterest.systems/development#Create-a-pull-request

Reply:
Agreed, probably don't need it if we're not doing components. However, if we do decide to extend a regular page with some interactivity, or we backtrack on component pages, this is pretty useful. Since they're super tiny functions, and pretty contained, I'm swaying in favor of keeping it

Reply:
Good idea, I'll re-do the development page in Markdown, and try adding it to that